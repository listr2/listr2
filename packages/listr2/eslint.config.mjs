import root from '../../eslint.config.mjs'
import { utils } from '@cenk1cenk2/eslint-config'

/** @type {import("eslint").Linter.Config[]} */
export default [...root, ...utils.configImportGroup({ tsconfigDir: import.meta.dirname, tsconfig: 'tsconfig.json' })]
