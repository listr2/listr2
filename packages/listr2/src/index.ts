/** @module listr2 */

export * from './listr'
export * from './constants'
export * from './presets'
export * from './interfaces'
export * from './renderer'
export * from './utils'
export { ListrEventManager, ListrTaskEventManager, EventManager, type Task as ListrTaskObject, type TaskWrapper as ListrTaskWrapper } from './lib'
