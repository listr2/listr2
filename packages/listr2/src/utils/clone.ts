import rfdc from 'rfdc'

const clone = rfdc({ circles: true })

/**
 * Deep clones a object in the easiest manner.
 */
// TODO: swap this with structured clone whenever we migrate to node 18
export function cloneObject<T extends Record<PropertyKey, any>>(obj: T): T {
  return clone(obj)
}
