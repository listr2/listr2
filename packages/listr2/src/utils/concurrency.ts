// adapted from https://github.com/fabiospampinato/promise-concurrency-limiter
// with MIT license, since it is again ESM only
// we needed to stop on the first failing case unlike the original
export class Concurrency {
  promise: Promise<void>

  private resolve: () => void
  private reject: (error: Error) => void
  private concurrency: number
  private count: number
  private queue: Set<() => void>

  constructor (options: { concurrency: number }) {
    this.concurrency = options.concurrency
    this.count = 0
    this.queue = new Set()
    this.initPromise()
  }

  public add<T>(fn: () => Promise<T>): Promise<T> {
    if (this.count < this.concurrency) {
      return this.run(fn)
    }

    return new Promise<T>((resolve) => {
      const callback = (): void => resolve(this.run(fn))

      this.queue.add(callback)
    })
  }

  private initPromise (): void {
    const { promise, resolve, reject } = this.withResolvers<void>()

    this.promise = promise
    this.resolve = resolve
    this.reject = reject
  }

  private flush (): void {
    for (const callback of this.queue) {
      if (this.count >= this.concurrency) {
        break
      }

      this.queue.delete(callback)

      callback()
    }
  }

  private run<T>(fn: () => Promise<T>): Promise<T> {
    this.count++

    const promise = fn()

    const cleanup = (): void => {
      this.count--

      this.flush()

      if (this.count === 0) {
        this.resolve()
      }
    }

    promise.then(cleanup, (error) => {
      this.queue.clear()
      this.reject(error)
    })

    return promise
  }

  private withResolvers<T>(): { promise: Promise<T>, resolve: (value: T) => void, reject: (error: Error) => void } {
    let resolve: (value: T) => void
    let reject: (error: Error) => void

    const promise = new Promise<T>((res, rej) => {
      resolve = res
      reject = rej
    })

    return {
      promise,
      resolve: resolve!,
      reject: reject!
    }
  }
}
