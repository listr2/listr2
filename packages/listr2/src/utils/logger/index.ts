export * from './logger.constants'
export type * from './logger.interface'
export * from './logger'
