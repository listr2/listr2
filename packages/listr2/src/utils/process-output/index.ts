export type * from './process-output-buffer.interface'
export * from './process-output-buffer'
export * from './process-output-stream'
export * from './process-output'
export type * from './process-output.interface'
export * from './writable'
