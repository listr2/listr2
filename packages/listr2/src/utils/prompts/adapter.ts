import { ListrTaskState } from '@constants'
import { PromptError } from '@interfaces'
import type { Task, TaskWrapper } from '@lib'

/* istanbul ignore next */
export abstract class ListrPromptAdapter {
  private state: ListrTaskState

  constructor(
    protected task: Task<any, any, any>,
    protected wrapper: TaskWrapper<any, any, any>
  ) {}

  protected reportStarted(): void {
    this.state = this.task.state

    if (this.task.prompt) {
      throw new PromptError('There is already an active prompt attached to this task which may not be cleaned up properly.')
    }
    this.task.prompt = this

    this.task.state$ = ListrTaskState.PROMPT
  }

  protected reportFailed(): void {
    this.task.state$ = ListrTaskState.PROMPT_FAILED
    this.restoreState()
  }

  protected reportCompleted(): void {
    this.task.state$ = ListrTaskState.PROMPT_COMPLETED
    this.restoreState()
  }

  protected restoreState(): void {
    this.task.prompt = undefined

    if (this.state) {
      // without pushing it through the subscriptions again, just set the state back to original
      this.task.state = this.state
    }
  }

  public abstract run<T = any>(...args: any[]): T | Promise<T>
}
