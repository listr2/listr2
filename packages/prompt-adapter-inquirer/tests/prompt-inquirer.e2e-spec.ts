import { AbortPromptError } from '@inquirer/core'
import { input } from '@inquirer/prompts'
import { Listr, delay } from 'listr2'

import { ListrInquirerPromptAdapter } from '@root'
import type { MockProcessOutput, RendererSetup } from '@tests/utils'
import { KEYS, RENDERER_SETUP, mockProcessOutput, unmockProcessOutput } from '@tests/utils'

describe.each<RendererSetup>(RENDERER_SETUP)('%s renderer: prompt -> inquirer', (renderer, rendererOptions) => {
  const output: MockProcessOutput = {} as MockProcessOutput

  process.stdout.isTTY = true

  beforeEach(async() => {
    mockProcessOutput(output)
  })

  afterEach(async() => {
    unmockProcessOutput(output)
    jest.clearAllMocks()
  })

  it('should render a prompt', async() => {
    const ctx = await new Listr(
      [
        {
          title: 'This task will execute.',
          task: async(ctx, task): Promise<void> => {
            const output = task.prompt(ListrInquirerPromptAdapter).run(input, {
              message: 'Give me some input.'
            })

            await delay(50)

            process.stdin.emit('data', Buffer.from(`test${KEYS.ENTER}`))

            ctx.output = await output
          }
        }
      ],
      {
        concurrent: false,
        renderer,
        rendererOptions
      }
    ).run()

    expect(ctx.output).toBe('test')
    // expectProcessOutputToMatchSnapshot(output, 'zTXlHkwllFGUiGXlWztXAeZMvIleiEFF')
  })

  it('should be able to cancel a prompt', async() => {
    let err: Error

    try {
      await new Listr(
        [
          {
            title: 'This task will execute.',
            task: async(_, task): Promise<void> => {
              const prompt = task.prompt(ListrInquirerPromptAdapter)

              void delay(50).then(() => prompt.cancel())

              await prompt.run(input, {
                message: 'Give me some input.'
              })
            }
          }
        ],
        {
          concurrent: false,
          renderer,
          rendererOptions
        }
      ).run()
    } catch (e) {
      err = e as Error
    }

    expect(err).toEqual(new AbortPromptError({ cause: 'Prompt was canceled' }))
    // expectProcessOutputToMatchSnapshot(output, 'AABZsFFYgZITgdWMQdLGmxABwlBaXdiv')
  })
})
