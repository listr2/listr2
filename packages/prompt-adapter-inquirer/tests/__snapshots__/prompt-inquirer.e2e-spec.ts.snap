// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`default renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-exit 1`] = `[]`;

exports[`default renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stderr 1`] = `[]`;

exports[`default renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stdout 1`] = `
[
  "[?25l",
  "◼ This task will execute.
",
  "[2K[1A[2K[G❯ This task will execute.
",
  "[2K[1A[2K[G❯ This task will execute.

? Give me some input.
",
  "[2K[1A[2K[1A[2K[1A[2K[G◼ This task will execute.

? Give me some input.
",
  "[2K[1A[2K[1A[2K[1A[2K[G✖ This task will execute.
",
  "[2K[1A[2K[G✖ Prompt was aborted
",
  "[2K[1A[2K[G",
  "✖ Prompt was aborted
",
  "[?25h",
]
`;

exports[`default renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-exit 1`] = `[]`;

exports[`default renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stderr 1`] = `[]`;

exports[`default renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stdout 1`] = `
[
  "[?25l",
  "◼ This task will execute.
",
  "[2K[1A[2K[G❯ This task will execute.
",
  "[2K[1A[2K[G❯ This task will execute.

? Give me some input.
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ This task will execute.

Give me some input. t
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ This task will execute.

Give me some input. te
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ This task will execute.

Give me some input. tes
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ This task will execute.

Give me some input. test
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ This task will execute.

✔ Give me some input. test
",
  "[2K[1A[2K[1A[2K[1A[2K[G◼ This task will execute.
",
  "[2K[1A[2K[G❯ This task will execute.
",
  "[2K[1A[2K[G✔ This task will execute.
",
  "[2K[1A[2K[G",
  "✔ This task will execute.
",
  "[?25h",
]
`;

exports[`simple renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-exit 1`] = `[]`;

exports[`simple renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stderr 1`] = `
[
  "? Give me some input.[23G",
  "
[?25h",
  "✖ This task will execute. [FAILED: Prompt was aborted]
",
]
`;

exports[`simple renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stdout 1`] = `
[
  "❯ This task will execute.
",
  "[?25l",
]
`;

exports[`simple renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-exit 1`] = `[]`;

exports[`simple renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stderr 1`] = `
[
  "? Give me some input.[23G",
  "[24G",
  "[2K[G? Give me some input. t[24G",
  "[25G",
  "[2K[G? Give me some input. te[25G",
  "[26G",
  "[2K[G? Give me some input. tes[26G",
  "[27G",
  "[2K[G? Give me some input. test[27G",
  "[23G",
  "[2K[G? Give me some input. test[27G",
  "[2K[G✔ Give me some input. test[27G",
  "
[?25h",
]
`;

exports[`simple renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stdout 1`] = `
[
  "❯ This task will execute.
",
  "[?25l",
  "[?25h",
  "✔ This task will execute.
",
]
`;

exports[`test renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-exit 1`] = `[]`;

exports[`test renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stderr 1`] = `
[
  "{"event":"MESSAGE","data":{"error":"Prompt was aborted"},"task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":true,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":false,"isStarted":false,"hasFailed":true,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
]
`;

exports[`test renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stdout 1`] = `
[
  "{"event":"STATE","data":"STARTED","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":true,"isStarted":true,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"STATE","data":"PROMPT","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"? Give me some input.\\u001b[23G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"STATE","data":"PROMPT_FAILED","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":false,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\n\\u001b[?25h","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":true,"isStarted":true,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"STATE","data":"PROMPT_FAILED","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":false,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"STATE","data":"FAILED","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":true,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":false,"isStarted":false,"hasFailed":true,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
]
`;

exports[`test renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-exit 1`] = `[]`;

exports[`test renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stderr 1`] = `[]`;

exports[`test renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stdout 1`] = `
[
  "{"event":"STATE","data":"STARTED","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":true,"isStarted":true,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"STATE","data":"PROMPT","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"? Give me some input.\\u001b[23G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[24G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[2K\\u001b[G? Give me some input. t\\u001b[24G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[25G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[2K\\u001b[G? Give me some input. te\\u001b[25G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[26G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[2K\\u001b[G? Give me some input. tes\\u001b[26G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[27G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[2K\\u001b[G? Give me some input. test\\u001b[27G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[23G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[2K\\u001b[G? Give me some input. test\\u001b[27G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\u001b[2K\\u001b[G✔ Give me some input. test\\u001b[27G","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"PROMPT","data":"\\n\\u001b[?25h","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":true,"isPaused":false,"isPending":true,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"STATE","data":"PROMPT_COMPLETED","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":false,"isSkipped":false,"hasFinalized":false,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":false,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"hasRolledBack":false,"isRollingBack":false,"isCompleted":true,"isSkipped":false,"hasFinalized":true,"hasSubtasks":false,"title":"This task will execute.","hasReset":false,"hasTitle":true,"isPrompt":false,"isPaused":false,"isPending":false,"isStarted":false,"hasFailed":false,"isEnabled":true,"isRetrying":false,"path":["This task will execute."]}}
",
]
`;

exports[`verbose renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-exit 1`] = `[]`;

exports[`verbose renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stderr 1`] = `
[
  "[FAILED] Prompt was aborted
",
]
`;

exports[`verbose renderer: prompt -> inquirer should be able to cancel a prompt: AABZsFFYgZITgdWMQdLGmxABwlBaXdiv-stdout 1`] = `
[
  "[STARTED] This task will execute.
",
  "[PROMPT] ? Give me some input.
",
]
`;

exports[`verbose renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-exit 1`] = `[]`;

exports[`verbose renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stderr 1`] = `[]`;

exports[`verbose renderer: prompt -> inquirer should render a prompt: zTXlHkwllFGUiGXlWztXAeZMvIleiEFF-stdout 1`] = `
[
  "[STARTED] This task will execute.
",
  "[PROMPT] ? Give me some input.
",
  "[PROMPT] Give me some input. t
",
  "[PROMPT] Give me some input. te
",
  "[PROMPT] Give me some input. tes
",
  "[PROMPT] Give me some input. test
",
  "[PROMPT] Give me some input. test
",
  "[PROMPT] ✔ Give me some input. test
",
  "[COMPLETED] This task will execute.
",
]
`;
